source/uwp/Renderer/lib/AdaptiveColumnRenderer.cpp

```diff
@@ -3,168 +3,135 @@
 #include "pch.h"
 
 #include "AdaptiveColumnRenderer.h"
+#include "AdaptiveColumnRenderer.g.cpp"
 
 #include "ActionHelpers.h"
 #include "AdaptiveRenderArgs.h"
 
-using namespace Microsoft::WRL;
-using namespace Microsoft::WRL::Wrappers;
-using namespace ABI::AdaptiveCards::Rendering::Uwp;
-using namespace ABI::AdaptiveCards::ObjectModel::Uwp;
-using namespace ABI::Windows::Foundation;
-using namespace ABI::Windows::Foundation::Collections;
-using namespace ABI::Windows::UI::Xaml;
-using namespace ABI::Windows::UI::Xaml::Controls;
-using namespace ABI::Windows;
-
-namespace AdaptiveCards::Rendering::Uwp
+namespace winrt::AdaptiveCards::Rendering::Uwp::implementation
 {
-    HRESULT AdaptiveColumnRenderer::RuntimeClassInitialize() noexcept { return S_OK; }
-
-    HRESULT AdaptiveColumnRenderer::Render(_In_ IAdaptiveCardElement* adaptiveCardElement,
-                                           _In_ IAdaptiveRenderContext* renderContext,
-                                           _In_ IAdaptiveRenderArgs* renderArgs,
-                                           _COM_Outptr_ IUIElement** ColumnControl) noexcept
-    try
+    rtxaml::UIElement AdaptiveColumnRenderer::Render(rtom::IAdaptiveCardElement const& cardElement,
+                                                     rtrender::AdaptiveRenderContext const& renderContext,
+                                                     rtrender::AdaptiveRenderArgs const& renderArgs)
     {
-        ComPtr<IAdaptiveCardElement> cardElement(adaptiveCardElement);
-        ComPtr<IAdaptiveColumn> adaptiveColumn;
-        RETURN_IF_FAILED(cardElement.As(&adaptiveColumn));
+        try
+        {
+            auto adaptiveColumn = cardElement.as<rtom::AdaptiveColumn>();
 
-        ComPtr<IBorder> columnBorder =
-            XamlHelpers::CreateABIClass<IBorder>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Border));
+            rtxaml::Controls::Border columnBorder{};
 
-        ComPtr<WholeItemsPanel> columnPanel;
-        RETURN_IF_FAILED(MakeAndInitialize<WholeItemsPanel>(&columnPanel));
+            auto columnPanel = winrt::make<rtrender::implementation::WholeItemsPanel>();
 
-        ComPtr<IUIElement> columnPanelAsUIElement;
-        RETURN_IF_FAILED(columnPanel.As(&columnPanelAsUIElement));
+            columnBorder.Child(columnPanel);
 
-        RETURN_IF_FAILED(columnBorder->put_Child(columnPanelAsUIElement.Get()));
+            auto previousContextRtl = renderContext.Rtl();
+            auto currentRtl = previousContextRtl;
+            auto containerRtl = adaptiveColumn.Rtl();
 
-        ComPtr<IFrameworkElement> columnPanelAsFrameworkElement;
-        RETURN_IF_FAILED(columnPanel.As(&columnPanelAsFrameworkElement));
+            bool updatedRtl = false;
 
-        ComPtr<IReference<bool>> previousContextRtl;
-        RETURN_IF_FAILED(renderContext->get_Rtl(&previousContextRtl));
-        ComPtr<IReference<bool>> currentRtl = previousContextRtl;
+            if (containerRtl != nullptr)
```

Comment:
the recommend way is `if (containerRtl)`
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es87-dont-add-redundant--or--to-conditions